app/menus/menu.js

```diff
@@ -17,6 +17,15 @@ const appVersion = app.getVersion();
 
 module.exports = (createWindow, updatePlugins, getLoadedPluginVersions) => {
   const commands = getKeymaps().commands;
+
+  const contextMenuTemplate = () => {
+    return editMenu(commands).submenu
+    .concat({type: 'separator'}, shellMenu(commands, createWindow).submenu);
+  };
+
+  // expose to plugins
+  app.contextMenuTemplate = contextMenuTemplate();
```

Comment:
It could be better to have same way than menu, to decorate it from plugins:
Plugins export a `decorateContextMenu` function which is called when app start like [here](https://github.com/zeit/hyper/blob/de8e70f935c79bb25d996ec53bd87ca7a525922c/app/index.js#L176) 

Or maybe call a plugin exported function when context menu will show for a given term. With this function, plugin can (un)check some item according to its internal state for this term. Very useful for plugin like `hyper-broadcast` or for a plugin that will let user define term style directly from context menu 😍